util/subnetworkid/subnetworkid_test.go

```diff
@@ -0,0 +1,443 @@
+			subnetworkID[:], buf)
+	}
+
+	// Ensure contents of SubnetworkID of block 234440 don't match 234439.
+	if subnetworkID.IsEqual(blockSubnetworkID) {
+		t.Errorf("IsEqual: SubnetworkID contents should not match - got: %v, want: %v",
+			subnetworkID, blockSubnetworkID)
+	}
+
+	// Set SubnetworkID from byte slice and ensure contents match.
+	err = subnetworkID.SetBytes(blockSubnetworkID.CloneBytes())
+	if err != nil {
+		t.Errorf("SetBytes: %v", err)
+	}
+	if !subnetworkID.IsEqual(blockSubnetworkID) {
+		t.Errorf("IsEqual: SubnetworkID contents mismatch - got: %v, want: %v",
+			subnetworkID, blockSubnetworkID)
+	}
+
+	// Ensure nil SubnetworkIDes are handled properly.
+	if !(*SubnetworkID)(nil).IsEqual(nil) {
+		t.Error("IsEqual: nil SubnetworkIDes should match")
+	}
+	if subnetworkID.IsEqual(nil) {
+		t.Error("IsEqual: non-nil SubnetworkID matches nil SubnetworkID")
+	}
+
+	// Invalid size for SetBytes.
+	err = subnetworkID.SetBytes([]byte{0x00})
+	if err == nil {
+		t.Errorf("SetBytes: failed to received expected err - got: nil")
+	}
+
+	// Invalid size for New.
+	invalidSubnetworkID := make([]byte, IDLength+1)
+	_, err = New(invalidSubnetworkID)
+	if err == nil {
+		t.Errorf("New: failed to received expected err - got: nil")
+	}
+}
+
+// TestSubnetworkIDString  tests the stringized output for SubnetworkIDes.
+func TestSubnetworkIDString(t *testing.T) {
+	wantStr := "ecaad478d2b00432346c3f1f3986da1afd33e506"
+	SubnetworkID := SubnetworkID([IDLength]byte{
+		0x06, 0xe5, 0x33, 0xfd, 0x1a,
+		0xda, 0x86, 0x39, 0x1f, 0x3f,
+		0x6c, 0x34, 0x32, 0x04, 0xb0,
+		0xd2, 0x78, 0xd4, 0xaa, 0xec,
+	})
+
+	SubnetworkIDStr := SubnetworkID.String()
+	if SubnetworkIDStr != wantStr {
+		t.Errorf("String: wrong SubnetworkID string - got %v, want %v",
+			SubnetworkIDStr, wantStr)
+	}
+}
+
+func TestSubnetworkIDesStrings(t *testing.T) {
+	first := SubnetworkID{
+		0x06, 0xe5, 0x33, 0xfd, 0x1a,
+		0xda, 0x86, 0x39, 0x1f, 0x3f,
+		0x6c, 0x34, 0x32, 0x04, 0xb0,
+		0xd2, 0x78, 0xd4, 0xaa, 0xec,
+	}
+	firstStr := "ecaad478d2b00432346c3f1f3986da1afd33e506"
+
+	second := SubnetworkID{}
+	secondStr := "0000000000000000000000000000000000000000"
+
+	tests := []struct {
+		name            string
+		SubnetworkIDes  []SubnetworkID
+		expectedStrings []string
+	}{
+		{"empty", []SubnetworkID{}, []string{}},
+		{"two SubnetworkIDes", []SubnetworkID{first, second}, []string{firstStr, secondStr}},
+		{"two SubnetworkIDes inversed", []SubnetworkID{second, first}, []string{secondStr, firstStr}},
+	}
+
+	for _, test := range tests {
+		strings := Strings(test.SubnetworkIDes)
+		if !reflect.DeepEqual(strings, test.expectedStrings) {
+			t.Errorf("SubnetworkIDesStrings: %s: expected: %v, got: %v",
+				test.name, test.expectedStrings, strings)
+		}
+	}
+}
+
+// TestNewFromStr executes tests against the NewFromStr function.
+func TestNewFromStr(t *testing.T) {
+	tests := []struct {
+		in   string
+		want SubnetworkID
+		err  error
```

Comment:
Please rename to expectedErr/wantErr or something that makes it clear this is what we expect